packages/embark-deployment/src/contract_deployer.js

```diff
@@ -316,7 +316,14 @@ class ContractDeployer {
         next();
       },
       function estimateCorrectGas(next) {
-        if (contract.gas === 'auto' || !contract.gas) {
+        if (contract._skipGasEstimations) {
```

Comment:
We have `currentContext` that is an array of the current contexts, so you could check for the `test` context in the array instead of that new flag? Or if we think the flag might be reused, we can keep it.

Reply:
That's a good watch, though IMO this (context) should be on the caller and it's preferable for the module not be aware of the  context

Reply:
I see what you're saying, but this really belongs in the caller. The contract deployer should really be environment agnostic.